cylc/flow/scripts/lint.py

```diff
@@ -0,0 +1,392 @@
+        },
+        # 2 or 4 space indentation both seem reasonable:
+        re.compile(r'^(\s|\s{3}|\s{5,})\[\[.*\]\]'): {
+            'short': 'wrong number of indents for second level section.',
+            'url': STYLE_GUIDE + 'indentation'
+        },
+        re.compile(r'^(\s{1,3}|\s{5,7}|\s{9,})\[\[\[.*\]\]\]'): {
+            'short': 'wrong number of indents for third level section.',
+            'url': STYLE_GUIDE + 'indentation'
+        },
+        re.compile(r'\s$'): {
+            'short': 'wrong number of indents for third level section.',
+            'url': STYLE_GUIDE + 'trailing-whitespace'
+        },
+        re.compile(r'^.{80,}'): {
+            'short': 'line > 79 characters.',
+            'url': STYLE_GUIDE + 'line-length-and-continuation'
+        },
+        re.compile(r'inherit\s*=\s*.*[a-z].*$'): {
+            'short': 'Family name contains lowercase characters.',
+            'url': STYLE_GUIDE + 'task-naming-conventions'
+        },
+
+    }
+}
+
+
+def parse_checks(check_arg):
+    """Collapse metadata in checks dicts.
+    """
+    parsedchecks = {}
+    if check_arg == '8':
+        purpose_filters = ['7-to-8']
+    elif check_arg == 'lint':
+        purpose_filters = ['lint']
+    else:
+        purpose_filters = ['lint', '7-to-8']
+
+    for purpose, checks in CHECKS.items():
+        if purpose in purpose_filters:
+            for index, (pattern, meta) in enumerate(checks.items()):
+                meta.update({'purpose': purpose})
+                meta.update({'index': index})
+                parsedchecks.update({pattern: meta})
+    return parsedchecks
+
+
+def check_cylc_file(file_, checks, modify=False):
+    """Check A Cylc File for Cylc 7 Config"""
+    # Set mode as read-write or read only.
+    outlines = []
+
+    # Open file, and read it's line to mempory.
+    lines = file_.read_text().split('\n')
+    count = 0
+    for line_no, line in enumerate(lines):
+        for check, message in checks.items():
+            if check.findall(line) and not line.strip().startswith('#'):
+                count += 1
+                if modify:
+                    if message['url'].startswith('http'):
+                        url = message['url']
+                    else:
+                        url = URL_STUB + message['url']
+                    outlines.append(
+                        f'# [{message["index"]:03d}: {message["purpose"]}]: '
+                        f'{message["short"]}\n'
+                        f'# - see {url}'
+                    )
+                else:
+                    print(
+                        Fore.YELLOW +
+                        f'[{message["index"]:03d}: {message["purpose"]}]'
+                        f'{file_}: {line_no}: {message["short"]}'
+                    )
+        if modify:
+            outlines.append(line)
+    if modify:
+        file_.write_text('\n'.join(outlines))
+    return count
+
+
+def get_cylc_files(base: Path) -> Generator:
```

Comment:
```suggestion
def get_cylc_files(base: Path) -> Generator[Path, None, None]:
```

https://docs.python.org/3/library/typing.html#typing.Generator